### DISCUSSION TITLE
JSON feed transition deadline in March 2025 -- has anyone done this yet?   What do we need to know?

### DISCUSSION THREAD

**Discussion by MetaMan2023** [amzn/selling-partner-api-models#4428] [Q&A]
I keep getting "warnings" in our inventory feeds (and other feeds) about a transition to JSON format in March 2025.

Has anyone done this?  Is it straight forward or are there any "gotchas"?

I'm mostly concerned about the inventory feeds -- we having been using the old MWS inventory feed XML files from like early 2000s and I'm concerned and wondering if the JSON transition is just a straight forward conversion to a JSON file (with the same data format) or if the feeds require a completely different base file and parameters?

Our entire system is built on the "variable" names and positions in the old XML data feeds.

Also what other parts of the system require this change to JSON?

Thanks

**Comment by SylvieManx** [comment]
Please refer to:
In-depth exploration of Amazon listing API (Part 1) - Article by Su Popian - Zhihu
https://zhuanlan.zhihu.com/p/705384610

**Comment by ritikacd** [comment]
Hi @MetaMan2023

You might have heard already that we are deprecating XML and flat file listings data (including pricing, inventory, relationships and images) submitted via the Feeds API. If you have an application that uses XML and flat file listings data, you have until March 31, 2025 to migrate to the JSON-based Listings Items APIs or the JSON_LISTINGS_FEED submitted via the Feeds API.  Refer to [this](https://developer-docs.amazon.com/sp-api/changelog/deprecation-of-feeds-api-support-for-xml-and-flat-file-listings-feeds) announcement to view feed types deprecating. 

We have developed multiple use case guide, videos and sample application to help you migrate your applications seamless. To get started we recommend following resources:
- [Migrating Listings Management Workflows Guide](https://developer-docs.amazon.com/sp-api/docs/migrating-listings-management-workflows) 
- [Listings XSD to JSON Migration Webinar](https://www.youtube.com/watch?v=MJHQPXWwGWo)

In the above migration guide, please refer to the [Migrating inventory feeds workflows](https://developer-docs.amazon.com/sp-api/docs/listing-workflow-migration-tutorial#migrating-inventory-feeds-workflows) section to see how to migrate your inventory feeds to the JSON based APIs. Please reach out if you have any questions. 

We hope this helps! 

Thanks,
Ritika
Selling Partner API Services

**Reply by J-W-Chan** [reply]
Do you have any official tested examples of converting json-schema to Vue forms? 
I have tried all the third-party frameworks here（ https://json-schema.org/tools )
Neither of them can effectively convert the json-schema returned by the Product Type Definitions API into an HTML form (due to missing required attribute rendering and dropdown data linkage effects).
We are currently unable to migrate. 
Please help us.
